luigi/contrib/beam_dataflow.py

```diff
@@ -0,0 +1,451 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright 2019 Spotify AB
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import abc
+from abc import abstractmethod, abstractproperty, ABCMeta
+import logging
+import json
+import os
+import subprocess
+
+import luigi
+from luigi import six
+from luigi.contrib import bigquery, gcs
+from luigi.task import MixinNaiveBulkComplete
+
+logger = logging.getLogger('luigi-interface')
+
+
+@six.add_metaclass(abc.ABCMeta)
+class DataflowParams(object):
```

Comment:
It would be clearer if this is named something like `DataflowParamKeys` or `DataflowParamNames` otherwise it sounds like it wraps the param values, instead of the param names/keys.